kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/metrics/MetricsCollector.java

```diff
@@ -345,8 +339,8 @@ private String collectMetrics(String metricsPodIp, String podName) throws Interr
     @SuppressWarnings("unchecked")
     public void collectMetricsFromPods() {
         Map<String, String>[] metricsData = (Map<String, String>[]) new HashMap[1];
-        TestUtils.waitFor("metrics to contain data", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
-                () -> {
+        await().atMost(Constants.GLOBAL_TIMEOUT).pollInterval(Constants.GLOBAL_POLL_INTERVAL)
+                .until(() -> {
```

Comment:
use the two arg form of until so you can access what was produced.  that gets rid of the that odd array of hashmaps.

(untested code)
```
        collectedData = await().atMost(Constants.GLOBAL_TIMEOUT).pollInterval(Constants.GLOBAL_POLL_INTERVAL)
                .until(this::collectMetricsFromPodsWithoutWait,
                        (collected) -> {
                            // KafkaExporter metrics should be non-empty
                            if (collectedData.isEmpty()) {
                                return false;
                            }

                            for (Map.Entry<String, String> item : collected.entrySet()) {
                                if (item.getValue().isEmpty()) {
                                    return false;
                                }
                            }
                            return true;
                        });
```

Reply:
you should be able to reduce the second block to something like:

```
   return !(collected.isEmpty() || collected.entrySet().stream().anyMatch(e -> e.getValue().isEmpty()));
```